### PR TITLE
[REVIEW] Add macros with SCOPED_TRACE for showing line of failure for cudf::test::expect_*

### PR DESCRIPTION
**Pull request by karthikeyann**
Add macros with SCOPED_TRACE for showing line of failure for `cudf::test::expect_*`

replace 
`cudf::test::expect_columns_equal` with `CUDF_TEST_EXPECT_COLUMNS_EQUAL`
`cudf::test::expect_columns_equivalent` with `CUDF_TEST_EXPECT_COLUMNS_EQUIVALENT`
`cudf::test::expect_tables_equal` with `CUDF_TEST_EXPECT_TABLES_EQUAL`
`cudf::test::expect_tables_equivalent` with `CUDF_TEST_EXPECT_TABLES_EQUIVALENT`

### CODE REVIEW

**Comment by GPUtester** [comment]
Please update the changelog in order to start CI tests.

View the gpuCI docs [here](https://docs.rapids.ai/gpuci#open-pr-or-update-pr).

File: cpp/tests/utilities/column_utilities.hpp [hunk 1]
```diff
@@ -201,16 +201,16 @@ inline std::pair<thrust::host_vector<std::string>, std::vector<bitmask_type>> to
...
 
-#define CUDF_TEST_EXPECT_COLUMNS_EQUAL(lhs, rhs) \
-  {                                              \
-    SCOPED_TRACE(" <--  line of failure\n");     \
-    cudf::test::expect_columns_equal(lhs, rhs);  \
+#define CUDF_TEST_EXPECT_COLUMNS_EQUAL(lhs, rhs...) \
+  {                                                 \
+    SCOPED_TRACE(" <--  line of failure\n");        \
+    cudf::test::expect_columns_equal(lhs, rhs);     \
   }
```
Comment (jrhemstad) [inline cpp/tests/utilities/column_utilities.hpp:208]: https://stackoverflow.com/questions/1067226/c-multi-line-macro-do-while0-vs-scope-block
```suggestion
  do {                                                 \
    SCOPED_TRACE(" <--  line of failure\n");        \
    cudf::test::expect_columns_equal(lhs, rhs);     \
  } while(0)
```

File: cpp/tests/utilities/column_utilities.hpp [hunk 2]
```diff
@@ -201,16 +201,16 @@ inline std::pair<thrust::host_vector<std::string>, std::vector<bitmask_type>> to
...
 
-#define CUDF_TEST_EXPECT_COLUMNS_EQUIVALENT(lhs, rhs) \
-  {                                                   \
-    SCOPED_TRACE(" <--  line of failure\n");          \
-    cudf::test::expect_columns_equivalent(lhs, rhs);  \
+#define CUDF_TEST_EXPECT_COLUMNS_EQUIVALENT(lhs, rhs...) \
+  {                                                      \
+    SCOPED_TRACE(" <--  line of failure\n");             \
+    cudf::test::expect_columns_equivalent(lhs, rhs);     \
   }
```
Comment (jrhemstad) [inline cpp/tests/utilities/column_utilities.hpp:214]:
```suggestion
#define CUDF_TEST_EXPECT_COLUMNS_EQUIVALENT(lhs, rhs...) \
  do {                                                      \
    SCOPED_TRACE(" <--  line of failure\n");             \
    cudf::test::expect_columns_equivalent(lhs, rhs);     \
  } while(0)
```

File: cpp/tests/utilities/column_utilities.hpp (hunk 1 above)
Comment (nvdbaranec) [reply cpp/tests/utilities/column_utilities.hpp:208]: +1 for this.

**Pull request merged**